### ISSUE TITLE
Restore windows position does not restore the desktop in which the window originated

### ISSUE THREAD

**Issue by karypid** [sienori/Tab-Session-Manager#913]
## Short description
When restoring a session, all windows open in the current desktop, even if the "Restore windows position" setting is enabled. I would expect each window to open in the virtual desktop where it originated from.  I am using Windows 10 and Windows 11.

## Steps to reproduce

1. Start browser with Tab Session Manager
2. Open at least two windows
3. Place the first window in Virtual Desktop 1
4. Place the second window in Virtual Desktop 2
5. Save the session
6. Restore the session

## Expected result
After step 6 I expected two windows to be created, one in Desktop 1 and one in Desktop 2.

## Actual result
Both windows  are created in Desktop 1.

## Platform information
* Platform (OS):  Windows
* Version of browser: Firefox 94.0.1 (64-bit)
* Version of Tab Session Manager: 

## Additional context
Chrome remembers the position of windows properly (i.e. it includes the desktop). It would be nice if this could be supported for Firefox as well.

**Comment by pjoc** [comment]
I have a similar requirement but I'm using the KDE desktop on Linux. Currently I use the X11 window manager but in time that will change to Wayland. I have multiple windows, each with multiple tabs, across several different desktops, where each desktop has a specific purpose. Having to move these windows around every time I log in is tedious and I would love to be able to avoid it.

**Comment by allanlaal** [comment]
this is a duplicate of #401

Linux workspaces are handled in: #422

**Comment by allanlaal** [comment]
Linux workspaces support is a must
native Firefox restore already recognizes workspaces

all different Linux Desktop Managers use the same open standard afaik


## use case

I use Linux workspaces like one uses tabs and browser windows (36 of them, a hardcoded limit in Mate Desktop)
each new project gets its own workspace
there I open up lots of browser windows with lots of tabs when I do research

often I want to save one of those sessions, but saving All windows means the session saved gets 35 workspaces worth of noise


## Solution

option to `save` and an option to `save & close` all windows & tabs on current workspace


## Alternative

lots of ridiculous manual labour

**Comment by mabra** [comment]
Thank you, I support this too!
Sometimes needing an hour to re-arrange the windows to their workspaces!

**Comment by pixeldesignstudio** [comment]
Pleeeeeeease add this feature. It would save a lot of time! :)

**Comment by sebibasti0815** [comment]
The "Restore window's position" option does not work in Chrome for multi monitor systems as well. Windows are randomly opened on different screens and often with random sizes. The setup is complicated, the monitor, 2x4k with 150% zoom, 1x2k with 100%.
